library/core/src/num/mod.rs

```diff
@@ -1605,19 +1605,39 @@ mod verify {
         }
     }
 
+    macro_rules! generate_unchecked_mul_harness {
```

Comment:
1. I wrote a general macro for `add`, `sub`, and `mul`, as their verification logic is the same. You could remove this and modify all `generate_unchecked_mul_harness` to `generate_unchecked_math_harness`.
2. I don't think `kani::assume` is required here because, per the official function contracts doc, Kani automatically inserts them according to the `requires` of `unchecked_mul`.

Reply:
Oh, i just saw. Is this added for performance? If so, please ignore me.

Reply:
Could you add a line of comment to indicate that this `mul` macro is added due to performance limitation?

Reply:
No worries. I can do it on my side.